network-config-analyzer/CalicoPolicyYamlParser.py

```diff
@@ -569,24 +572,32 @@ def _set_affects_ingress_egress(self, policy_spec, is_profile, res_policy):
         if not res_policy.affects_egress and 'egress' in policy_spec:
             self.syntax_error('A NetworkPolicy with egress field but no "Egress" in its policyTypes', policy_spec)
 
-    def _set_selected_peers(self, policy_spec, is_profile, res_policy):
+    def _get_selected_peers(self, policy_spec, is_profile, is_global_np, policy_name):
         """
-        Set the selected_peers member of the policy according to the spec
+         Returns a set of peers which are selected by the current policy according to its spec
         :param dict policy_spec: The spec part in the yaml
         :param bool is_profile: Whether we parse a Profile object
-        :param CalicoNetworkPolicy res_policy: The NetworkPolicy to update the selected_peers for
-        :return: None
+        :param bool is_global_np: whether we parse a GlobalNetworkPolicy
+        :param str policy_name: The current policy name
+        :return: set of selected peers
+        :rtype: PeerSet
         """
+        peers = self.peer_container.get_all_peers_group()  # default of selector/ ns_selector is all()
         pod_selector = policy_spec.get('selector')
         if pod_selector:
             if is_profile:
                 self.syntax_error('selector is not allowed in the spec of a Profile', policy_spec)
-            res_policy.selected_peers = self._parse_label_selector(pod_selector, policy_spec, self.namespace)
+            peers &= self._parse_label_selector(pod_selector, policy_spec, self.namespace)
         else:
             if is_profile:
-                res_policy.selected_peers = self.peer_container.get_profile_pods(res_policy.name, True)
-            else:
-                res_policy.selected_peers = self.peer_container.get_namespace_pods(self.namespace)
+                return self.peer_container.get_profile_pods(policy_name, True)
+            elif not is_global_np:
+                return self.peer_container.get_namespace_pods(self.namespace)
```

Comment:
```suggestion
            else:
                peers = self.peer_container.get_namespace_pods(self.namespace)
```
I think we can apply this change and avoid having to check if this is global network policy.